compiler/packages/babel-plugin-react-compiler/src/__tests__/fixtures/compiler/capturing-func-mutate-2.expect.md

```diff
@@ -27,7 +27,6 @@ export const FIXTURE_ENTRYPOINT = {
 import { c as _c } from "react/compiler-runtime";
 function component(a, b) {
   const $ = _c(2);
-  const y = { b };
```

Comment:
I wondered about the exact mechanism that was kicking in to let us prune this. We have always had code that recursively calls DCE on inner functions (its part of AnalyzeFunctions). And DCE prunes unused context variables on the outermost function its called on. So we were pruning the context access for `y`. But because we didn't prune unused _deps_, the outer DCE thought that `y` was still used. Now the inner DCE removes y from context, and then the outer DCE can see that `y` is unused and prune it.

Reply:
Yep that's completely right. I thought about writing a separate pass to prune unused `deps` and their source instructions to reduce noise of the snapshot diffing for this PR. Ultimately it didn't seem worth the time